### PR TITLE
User: Donate tokens back to Faucet

### LINKED ISSUE #4: A09 User: Donate tokens back to Faucet
**Description:**
BSV Testnet tokens are limited supply, provide a mechanism to allow deposits to return unused coins.

**User story:**
As a user, I want to donate my unused or extra testnet coins back to the master treasury account, so that I can contribute to the faucet’s sustainability and reduce excess coins in circulation

**Acceptance criteria:**
- I can access the "Donate to Faucet" option in the faucet interface inside the /dashboard section. 
- The system displays a predefined, verified master wallet address to which users can send their testnet coins.
- I can select the amount of testnet coins I want to donate from my available balance.
- After confirming the donation, the system generates a transaction that sends the selected amount to the master treasury faucet address.
- A success message, along with the transaction ID, is shown after the coins are donated (in my /requests history page)
- I can track the donation transaction on the testnet explorer via a txid testnet blockchain explorer link.

### PR DESCRIPTION
**Pull request by ikemHood**
May Fix #4

### CODE REVIEW

**Comment by mo-jaber** [comment]
Hi @ikemHood can you rework this to ensure it fits with the latest baseline, as we have made some changes today

**Comment by ikemHood** [comment]
Alright

**Comment by mo-jaber** [comment]
Hi @ikemHood please update to reflect latest branch. the dependency https://github.com/bitcoin-sv/bsv-faucet/issues/17 is in

**Comment by ikemHood** [comment]
Screenshot from 2024-11-03 23-06-35

https://github.com/user-attachments/assets/68b5e3bd-da8b-41f7-90b1-c4ea1179a358

what I ahve so far

**Comment by ikemHood** [comment]
I Currently need to implement a ConnectWallet system to allow us get users balance from wallet and send transaction from their wallet, can i be pointed to what package or a docs on how i can implement it?

**Comment by mo-jaber** [comment]
Hey @ikemHood how are you going?

File: .env.example
Comment (mo-jaber) [inline .env.example:1]: pull latest version of this file.

File: app/(dashboard)/dashboard/page.tsx
Comment (mo-jaber) [inline app/(dashboard)/dashboard/page.tsx:1]: pull latest version of this file.

File: app/(dashboard)/donate/page.tsx
Comment (mo-jaber) [inline app/(dashboard)/donate/page.tsx:1]: as discussed. donate can go bottom of /dashboard.

File: app/(dashboard)/page.tsx
Comment (mo-jaber) [inline app/(dashboard)/page.tsx:1]: pull latest version of this app/(dashboard)/page.tsx

File: app/api/donation/route.ts
Comment (mo-jaber) [inline app/api/donation/route.ts:1]: as discussed, bottom of app/(dashboard)/page.tsx

File: package.json
Comment (mo-jaber) [inline package.json:1]: ignore git push in next PR, to avoid conflicts

**Comment by ikemHood** [comment]
Screenshot from 2024-11-05 15-13-16

**Comment by mo-jaber** [comment]
Thanks @ikemHood looks good.

**Pull request merged**